File: 백준/Silver/11561. 징검다리/징검다리.py
import sys
import math

def max_jumps(n):
    if n < 1:
        return 0
    k = int((-1 + math.sqrt(1 + 8 * n)) / 2)
    return k

n = int(sys.stdin.readline().strip())
arr = []
for _ in range(n):
    arr.append(int(sys.stdin.readline().strip()))

for i in arr:
    print(max_jumps(i))

File: 백준/Silver/11561. 징검다리/징검다리.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {

    public static int maxK(long N) {
        return (int)((-1 + Math.sqrt(1 + 8.0 * N)) / 2);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int num = Integer.parseInt(br.readLine().trim());

        long[] inputs = new long[num];
        for (int i = 0; i < num; i++) inputs[i] = Long.parseLong(br.readLine().trim());

        StringBuilder sb = new StringBuilder();
        for (long n : inputs) sb.append(maxK(n)).append("\n");

        System.out.print(sb);
    }
}